localstack/utils/aws/aws_stack.py

```diff
@@ -318,11 +319,17 @@ def connect_to_service(
         # identify requests as such
         if client and internal:
 
-            def _add_internal_header(request, **kwargs):
-                request.headers.add_header(HEADER_LOCALSTACK_ACCOUNT_ID, get_aws_account_id())
+            def _add_internal_header(account_id: str, request, **kwargs):
+                request.headers.add_header(HEADER_LOCALSTACK_ACCOUNT_ID, account_id)
+
+            # The handler invocation happens in boto context leading to loss of account ID
+            # Hence we build a partial here with the account ID baked-in.
+            _handler = functools.partial(
+                _add_internal_header, kwargs.get("aws_access_key_id", get_aws_account_id())
+            )
```

Comment:
Fixed this bug where `get_aws_account_id()` would run in a different context, presumably in boto, and would not return the correct account ID. Fixing by hardingcoding the account ID as a function partial.

Unfortunately this is not used anymore as the ability to receive origin account ID is dropped (in this PR) and will be addressed in the aws_stack rework project. This wasn't being anywhere so no functionality is impacted.

Reply:
After being registered in the event system (below), the handler is called with args:  `request` and `**kwargs`. There's no way to pass the account ID as an additional args to the handler when it is invoked. Previously the account ID would be retrieved from the context (incorrectly). Hence, now we simply bake in the account ID to the function instead